dockstore-client/src/main/java/io/dockstore/client/cli/Client.java

```diff
@@ -345,7 +347,15 @@ private static void downloadURL(String browserDownloadUrl, String installLocatio
             // Run the dockstore script with the 'self-install' argument so if it needs
             // to download the appropriate CLI JAR file it will do so now
             // and not the next time the user runs the dockstore script
-            Utilities.executeCommand(file.toPath().toString() + " self-install");
+            // Also set the Dockstore version environment variable to make
+            // sure that the dockstore script will download the new version of
+            // the client jar. The Dockstore version environment variable might
+            // be set to the previous version of Dockstore if the dockstore script
+            // was run to upgrade the Dockstore version.
+            Map<String, String> additionalEnvVarMap = new HashMap<>();
```

Comment:
This is fine, but you could simplify it to `Map<String, String> additionalEnvVarMap = Collections.singletonMap("DOCKSTORE_VERSION", dockstoreVersion);` and get rid of the next line.

Reply:
changed to incorporate your suggestion